WIP: Add @material-ui/icons support

gm666q opened the pull request:
This PR is work in progress.

Related issue: #25

subroh0508 committed:
[update]Update libraries

subroh0508 committed:
[add]Last 平成 commit

subroh0508 committed:
[add]First 令和 commit

subroh0508 committed:
[add]Add Breakpoints/Transitions/ZIndex

subroh0508 committed:
[edit]Replace new theme classes and methods

subroh0508 committed:
[delete]Delete unused classes

subroh0508 committed:
[delete]Delete unused classes

subroh0508 committed:
[edit]Add setter methods with block arguments

subroh0508 committed:
[edit]Change AppBar/Paper inherited RClass

subroh0508 committed:
[edit]Change Card inherited RClass

subroh0508 committed:
[edit]Delete old PaperElementBuilder

subroh0508 committed:
[edit]Change ButtonBase inherited RClass

subroh0508 committed:
[edit]Change RTransitionBuilder abstract and has RClass

subroh0508 committed:
[edit]Change RTransitionGroupBuilder having RClass

subroh0508 committed:
[edit]Rename CollapseType -> CollapseStyle

subroh0508 committed:
[edit]Change ClickAwayListenerBuilder having RClass

subroh0508 committed:
[edit]Change some ElementBuilders having RClass(A - C)

subroh0508 committed:
[edit]Change some ElementBuilders having RClass(D - I)

subroh0508 committed:
[edit]Change some ElementBuilders having RClass(L - R)

subroh0508 committed:
[edit]Change some ElementBuilders having RClass(S)

subroh0508 committed:
[edit]Change some ElementBuilders having RClass(T)

subroh0508 committed:
[edit]Move HeadlineMapping to typography.kt

subroh0508 committed:
[edit]Add TabIndicatorElementBuilder and enable tabIndicatorProps method

subroh0508 committed:
[edit]Delete Old MaterialElementBuilder

subroh0508 committed:
[add]Add StylesSet interface

subroh0508 committed:
v0.1.0

subroh0508 committed:
[update]js-wrappers: 1.0.0-pre.70-kotlin-1.3.21/react: 16.6.0-pre.70-kotlin-1.3.21

subroh0508 committed:
v0.1.1

subroh0508 committed:
[edit]Add childWithStyles method

subroh0508 committed:
v0.1.2

subroh0508 committed:
[edit]Add css setter method

subroh0508 committed:
[edit]Add TabStyle

subroh0508 committed:
v0.1.3

subroh0508 committed:
v0.1.5

subroh0508 committed:
[fix]Insert 'operator' prefix to GreyColors.get()

subroh0508 committed:
[fix]Move p argument on typography method

subroh0508 committed:
v0.1.6

subroh0508 committed:
v0.1.7

subroh0508 committed:
[fix]Change default 'components' attribute button to div on ExpansionPanelSummary

subroh0508 committed:
v0.1.9

subroh0508 committed:
v0.1.10(update libraries)

subroh0508 committed:
[fix]ListItemElementBuilder Tag -> T: Tag

subroh0508 committed:
v0.1.11

subroh0508 committed:
[edit]Add ButtonListItemElementBuilder

subroh0508 committed:
[fix]Fix ButtonListItemElementBuilder's style

subroh0508 committed:
v0.1.13

subroh0508 committed:
[fix]Replace function buildElement to rFunction in childWithStyles

subroh0508 committed:
v0.1.14

subroh0508 committed:
[fix]Add toString method

subroh0508 committed:
v0.1.15

subroh0508 committed:
v0.1.16(Update libraries)

subroh0508 committed:
[fix]Fix wrong library version

subroh0508 committed:
v0.1.17

subroh0508 committed:
[update]Update Gradle version 4.4 -> 5.3.1

subroh0508 committed:
[add]Add buildSrc module

subroh0508 committed:
[edit]Change scripts Groovy to Kotlin DSL

subroh0508 committed:
[edit]Delete comment

subroh0508 committed:
v0.2.0

subroh0508 committed:
[edit]Add a badge to README.md

subroh0508 committed:
[update]Update Kotlin -> 1.3.50

subroh0508 committed:
[edit]nodejs() -> browser()

subroh0508 committed:
v0.2.3

subroh0508 committed:
[add]Add webpackTask settings

subroh0508 committed:
[update]Update Kotlin -> 1.3.60

subroh0508 committed:
[edit]archiveFileName -> outputFileName

subroh0508 committed:
[update]Gradle -> 6.0.1

subroh0508 committed:
Merge pull request #3 from subroh0508/update-kotlin-1.3.60

Update kotlin 1.3.60(fix #2 )

subroh0508 committed:
[edit]Add .idea/artifacts to .gitignore

subroh0508 committed:
v0.2.4

subroh0508 committed:
[fix]Fix duplicated dependency(fix #8)

subroh0508 committed:
[fix]Remove settings.gradle.kts in sub-modules(fix #4)

subroh0508 committed:
[fix]Fix library dependencies

subroh0508 committed:
v0.2.5

subroh0508 committed:
v0.2.6

subroh0508 committed:
[update]Update Material-UI -> 4.8.3

subroh0508 committed:
[fix]Fix spacing.unit property -> spacing function

subroh0508 committed:
[fix]Remove color property from TypographyStyle

subroh0508 committed:
[fix]Rename MuiThemeProvider -> ThemeProvider

subroh0508 committed:
[fix]Fix spacing function

zole40 committed:
implemented breadcrumbs wrapper

zole40 committed:
added missing Divider properties

zole40 committed:
Implemented TreeView and TreeItem wrappers

zole40 committed:
Added align to grid, added new anchorEl tag to Popover

subroh0508 committed:
[update]Update Kotlin -> 1.3.70 / React -> 16.13.0

subroh0508 committed:
v0.2.7

subroh0508 committed:
Merge remote-tracking branch 'remotes/origin/master' into update-material-ui-v4

# Conflicts:
#	buildSrc/src/main/java/Versions.kt

subroh0508 committed:
[update]Update Material-UI -> 4.9.6

subroh0508 committed:
[fix]Fix spacing function(one argument)

subroh0508 committed:
[fix]Fix Grid spacing 0..10

subroh0508 committed:
[fix]Fix CardActions

subroh0508 committed:
[fix]Fix DialogActions

subroh0508 committed:
[fix]Fix Divider

subroh0508 committed:
[fix]Fix ExpansionPanel / ExpansionPanelActions

subroh0508 committed:
[fix]Fix SwitchStyle

subroh0508 committed:
[fix]Fix SvgIcon

subroh0508 committed:
[fix]Fix Tab

subroh0508 committed:
[fix]Fix TableCell

subroh0508 committed:
[fix]Fix InputLabel

subroh0508 committed:
[fix]Fix InputBase

subroh0508 committed:
[fix]Fix Typography

subroh0508 committed:
[fix]Fix index.html

subroh0508 committed:
[fix]Fix sample

subroh0508 committed:
Merge pull request #9 from subroh0508/update-material-ui-v4

Update Material-UI to v4

subroh0508 committed:
v0.3.0

subroh0508 committed:
[add]Add makeStyles method

subroh0508 committed:
v0.3.1

subroh0508 committed:
[fix]Fix Minxins properties number to number.px

subroh0508 committed:
v0.3.2

subroh0508 committed:
[add]Add AlertElementBuilder and support methods

subroh0508 committed:
[add]Add :lab module

subroh0508 committed:
[edit]Move AlertElementBuilder to :lab module

subroh0508 committed:
[Add]Add AlertTitle

subroh0508 committed:
[fix]Fix method name

subroh0508 committed:
[fix]Add onClose property to AlertProps

subroh0508 committed:
[update]Update Material-UI -> 4.9.8

subroh0508 committed:
Core: v0.3.3 / Lab: v0.3.0

subroh0508 committed:
[edit]Update README.md

subroh0508 committed:
[edit]Update README.md

oxc committed:
[add]Add ButtonGroup

subroh0508 committed:
[fix]Fix README.md

oxc committed:
[add]Add Skeleton

subroh0508 committed:
Merge pull request #10 from oxc/buttongroup

Add ButtonGroup

oxc committed:
[add]Add withStyles function that returns a RClass

Creating a new RClass in each invocation can prevent reconciliation of
such components. Extract a separate method that returns the styled
class, so it can be stored for recurring usage.

subroh0508 committed:
Merge pull request #11 from oxc/skeleton

Add Skeleton (labs)

subroh0508 committed:
[fix]Fix warning of sample(fixes #6, #7)

subroh0508 committed:
Merge pull request #12 from oxc/withStyles

[add]Add withStyles function that returns a RClass

oxc committed:
[fix]Fix withStyles with RPureComponent

The react.rClass extension is defined on RComponent, but should also
work for RPureComponent.
Fall back to manually casting to RClass, so it works for both.

subroh0508 committed:
Merge pull request #13 from oxc/withStyles_component

[fix]Fix withStyles with RPureComponent

subroh0508 committed:
Core: v0.3.4 / Lab: v0.3.1

oxc committed:
Deprecate childWithStyles

The childWithStyles variants always create a new component on each
invocation, which prevents reconciliation of such components.

Instead, withStyles should be used to store the created higher order
component in a RClass variable and call that to render the component.

subroh0508 committed:
Merge pull request #14 from oxc/deprecate_childWithStyles

Deprecate childWithStyles

oxc committed:
[fix]Update button props/styles

subroh0508 committed:
Merge pull request #15 from oxc/fix_button

Update Button to latest version

subroh0508 committed:
[add]add makeStyles with props argument

subroh0508 committed:
[edit]Change StylesBuilder receiving lambda

subroh0508 committed:
[delete]Delete StylesSet interface

subroh0508 committed:
[edit]Remove internal from StylesBuilder#css property

subroh0508 committed:
v0.3.6

oxc committed:
[fix] Fix buttons demo (remove outdated import)

subroh0508 committed:
[update]Update Kotlin -> 1.3.72

subroh0508 committed:
Merge pull request #16 from oxc/fix_buttons_demo

[fix] Fix buttons demo (remove outdated import)

subroh0508 committed:
v0.3.7

oxc committed:
[update]Update to wrapper pre.98

oxc committed:
[change]Use KClass.rClass in withStyles

With kotlin-react pre.98 this is now also available for Component

oxc committed:
[fix]Update typography to latest version

oxc committed:
[add]Add convenience functions for typography h1-h6

subroh0508 committed:
[edit]Add edge property to IconButtonProps

subroh0508 committed:
[add]Add useMediaQuery function

subroh0508 committed:
Merge pull request #18 from oxc/kotlin_wrappers_pre98

Update kotlin-wrappers to pre.98

subroh0508 committed:
Merge pull request #19 from oxc/typography

Update typography, add convenience

subroh0508 committed:
[update]Update kotlin-wrappers -> pre.100

subroh0508 committed:
v0.3.9

zole40 committed:
created Alert components

subroh0508 committed:
[fix]Fix Transitions#create method

subroh0508 committed:
[fix]Move parsing transitions method to TransitionsDelegate

subroh0508 committed:
v0.3.10

subroh0508 committed:
[add]Add useTheme

subroh0508 committed:
v0.3.11

subroh0508 committed:
[add]Add Container component

subroh0508 committed:
v0.3.12

subroh0508 committed:
[edit]Enable classes property on CssBaseline

subroh0508 committed:
Revert "[edit]Enable classes property on CssBaseline"

This reverts commit 603c88b0

subroh0508 committed:
[edit]Enable setting global styles

subroh0508 committed:
v0.3.13

zole40 committed:
fixed alert tag

zole40 committed:
fixed requested changes https://github.com/subroh0508/kotlin-material-ui/pull/22

zole40 committed:
reverted PopoverElementBuilder.kt changes

subroh0508 committed:
[add]Add :pickers module

subroh0508 committed:
[fix]Fix Link component P to A

subroh0508 committed:
[edit]Add focusVisible entry

subroh0508 committed:
v0.3.14

zole40 committed:
fixed requested changes https://github.com/subroh0508/kotlin-material-ui/pull/22

zole40 committed:
fixed ElementBuilders

zole40 committed:
clear imports

subroh0508 committed:
Merge pull request #22 from zole40/Contribution

Contribution

subroh0508 committed:
[edit]Add property setter extension method to TreeViewElementBuilder

subroh0508 committed:
[edit]Add property setter extension method to TreeItemElementBuilder

subroh0508 committed:
[edit]Add property setter extension method to BreadcrumbsElementBuilder

subroh0508 committed:
v0.3.15

oxc committed:
[update]Update kotlin-wrappers -> pre.104

kotlin-wrappers now exports all its npm dependencies as api.
Export @material-ui dependencies as well, and export all kotlin
dependencies that are part of the libraries' public API.
Remove superfluous dependency declarations where applicable.

subroh0508 committed:
Merge pull request #23 from oxc/kotlin_wrappers_pre104

[update]Update kotlin-wrappers -> pre.104

subroh0508 committed:
[add]Create BasePickerProps and BasePickerElementBuilder

subroh0508 committed:
[add]Add Tag extension properties to BasePickerElementBuilder

subroh0508 committed:
[add]Add date-io modules

subroh0508 committed:
[add]Add pickers:date-io:core module

subroh0508 committed:
[add]Add DateIOFormats and IUtils by external interface

subroh0508 committed:
[edit]Move :date-io:core -> :date-io

subroh0508 committed:
[add]Add DateFnsUtils

subroh0508 committed:
v0.3.16

subroh0508 committed:
[add]Add LocalizationProviderBuilder

subroh0508 committed:
[add]Add MobileDatePickerElementBuilder

subroh0508 committed:
Merge branch 'master' into add-material-ui-pickers-support

# Conflicts:
#	.idea/compiler.xml
#	buildSrc/src/main/java/Libraries.kt
#	sample/build.gradle.kts

subroh0508 committed:
[edit]Add DateValidationProps and fix accessing LocalizationProvider

subroh0508 committed:
[add]Add StaticWrapperElementBuilder

subroh0508 committed:
[add]Add DesktopDatePickerElementBuilder

subroh0508 committed:
[add]Add DatePickerElementBuilder

subroh0508 committed:
[add]Add DateInputProps/ArrowSwitcherElement/CalendarElement

subroh0508 committed:
[edit]Change BasePickerElementBuilder inherited RElementBuilder

subroh0508 committed:
[add]Add CalendarElementBuilder

subroh0508 committed:
[edit]Change second argument Array<String> -> dynamic onNodeSelect(fix #24)

subroh0508 committed:
[edit]Divide TreeViewElementBuilder to SingleSelect/MultiSelect(fix #24)

subroh0508 committed:
[Add]Add SliderElementBuilder(fix #26)

subroh0508 committed:
[add]Add ClockViewElementBuilder

subroh0508 committed:
[Add]Add TimePickerElementBuilders

subroh0508 committed:
[add]Add SharedPickerProps' properties to BasePickerProps

subroh0508 committed:
[fix](var|fun) Tag. -> (var|fun) RProps

subroh0508 committed:
[add]Add DateTimePickerElementBuilders

subroh0508 committed:
[edit]Inherit TextFieldProps of BasePickerProps

subroh0508 committed:
[edit]Remove unnecessary extension properties

subroh0508 committed:
v0.3.17

subroh0508 committed:
[Add]Add DateRangePickerElementBuilders

subroh0508 committed:
[edit]datePicker -> dateRangePicker

subroh0508 committed:
[Add]Add dateRangeDelimiter

subroh0508 committed:
[update]Update Material-UI -> 4.9.14 / Material-UI-Pickers -> 4.0.0-alpha.7

subroh0508 committed:
[edit]Change renderInput used buildElement -> buildElements

subroh0508 committed:
[edit]Add props method to MaterialElementBuilder

subroh0508 committed:
[Add]Add DayElementBuilder

subroh0508 committed:
[edit]Remove unnecessary extension method

subroh0508 committed:
Merge pull request #29 from subroh0508/add-material-ui-pickers-support

Add Material-UI pickers support

subroh0508 committed:
[edit]Fix bintray.gradle

subroh0508 committed:
v0.4.0

subroh0508 committed:
[edit]Aggregate @JsModule to imports.kt

subroh0508 committed:
[edit]Change Spacing to dynamic

subroh0508 committed:
[revert]Revert main.kt

subroh0508 committed:
[edit]Aggregate @JsModule to imports.kt(styles)

subroh0508 committed:
[edit]Replace [a-zA-Z]+Component to [a-zA-Z]+ defined imports.kt

subroh0508 committed:
[edit]Replace style methods to raw style method in imports.kt

subroh0508 committed:
[edit]Replace ThemeProvider to imports.kt

subroh0508 committed:
[update]Update @material-ui/core -> v4.10.0

subroh0508 committed:
v0.4.1

subroh0508 committed:
[edit]add warning/info/success property to Palette

subroh0508 committed:
v0.4.2

subroh0508 committed:
[add]Add ToggleButtonElementBuilder(fix #30)

subroh0508 committed:
[add]Add ToggleButtonGroupElementBuilder(fix #30)

subroh0508 committed:
v0.4.3

nikanorov committed:
update to more recent wrapper builds.

some old ones are unavailable already

nikanorov committed:
bump gradle version to support JDK14

subroh0508 committed:
Merge pull request #31 from nikanorov/master

update to more recent wrapper builds.

subroh0508 committed:
v0.4.4

subroh0508 committed:
[update]Update Material-UI -> v4.11.0

subroh0508 committed:
[update]Update kotlin-wrappers -> pre.110

subroh0508 committed:
v0.4.5

subroh0508 committed:
[update]Update Kotlin -> 1.4.0

subroh0508 committed:
[fix]Add external modifier

subroh0508 committed:
v0.5.0-beta1

subroh0508 committed:
[edit]Change IR backend IR -> BOTH

subroh0508 committed:
[edit]Add @JsNonModule annotation

subroh0508 committed:
Merge pull request #32 from subroh0508/kotlin-1.4.0

Update Kotlin to 1.4.0

gm666q committed:
Initial icons work

toliner commented:
Thanks for your work. Code in this PR help me a lot.

I think that kotlin wrapper of `@material-ui/icons` should be auto-generated, because there are over 1100 icons.

From [material-ui's document](https://material-ui.com/components/icons/)
> Material Design has standardized over 1,100 official icons, each in five different "themes" (see below). For each SVG icon, we export the respective React component from the @material-ui/icons package. You can search the full list of these icons.

Hand-writing such a large number of wrappers is so hard. 
I have an idea how to generate wrapper code:

1. Get list of icons from [material-icons's source dir](https://github.com/mui-org/material-ui/tree/next/packages/material-ui-icons/src)
1. Generate `imports.kt ` based on the list
1. Generate dsl functions (extension functions for `RBuilder`) for each icons

Here is the example of code-generation, build and publish: https://github.com/justincase-jp/AWS-CDK-Kotlin-DSL
I did it, so I can help you if you want to do code-generation.

---
Japanese:
英語が怪しいので日本語で要約

公式ドキュメントによると、Materialのアイコン数は1100以上、スタイルの差も含めたらjsファイルが6672個あるらしい。
手で書くのは流石に無理では？コード生成した方が良いと思います。
Material-uiのソースコードからiconの一覧は取得可能なので、それを使えばできるはずです。
私は別件でKotlinのコード生成 & ビルド & デプロイをやった事があるので、お手伝いできると思います。

subroh0508 commented:
@toliner Thanks for the suggestion! If you can write to write generator, I welcome your contribution😎

So please send me a Pull Request!

---

Japanese:

提案ありがとうございます！自動生成での対応、良いと思います😎
レビューは適宜しますので、Pull Requestを送ってくださると大変嬉しいです！

gm666q committed:
Add generated icons and update .gitignore

gm666q commented:
I wrote a crude generator in Rust I can share that or rewrite it if you prefer another language. I added remaining icons but didn't test it too much since there is more then 6000 icons. I also updated .gitingore because it wasn't ignoring OS specific files and gradle files correctly ([https://github.com/JetBrains/idea-gitignore](https://github.com/JetBrains/idea-gitignore)).
